apis/apiextensions/v1beta1/composition_types.go

```diff
@@ -133,14 +206,36 @@ type ReadinessCheck struct {
 	MatchInteger int64 `json:"matchInteger,omitempty"`
 }
 
-// Patch is used to patch the field on the base resource at ToFieldPath
-// after piping the value that is at FromFieldPath of the target resource through
-// transformers.
+// PatchSet defines a named list of patches on a composition.
+type PatchSet struct {
+	ID      string  `json:"id"`
+	Patches []Patch `json:"patches"`
+}
+
+// patchSetMap type stores a list of patches from a set, against the set ID.
+type patchSetMapType map[string][]Patch
+
+// PatchType is type of the patch to be chosen.
+type PatchType string
+
+// Accepted TransformTypes.
+const (
+	PatchTypeStandard  PatchType = "std"
+	PatchTypeReference PatchType = "ref"
+)
```

Comment:
We have a handful of additional use cases where patch types could be used - I'd like to make sure we consider all of them before we introduce any. I'll write up an issue tracking the ones I know of so we can make sure to pick types that will be as future proof as possible.

Reply:
Here's that issue: https://github.com/crossplane/crossplane/issues/1974